sdk/lib/queue/queue_compartment.cc

```diff
@@ -77,105 +83,106 @@ int queue_destroy_sealed(Timeout            *timeout,
                          AllocatorCapability heapCapability,
                          CHERI_SEALED(MessageQueue *) queueHandle)
 {
-	if (MessageQueue *queue = unseal(queueHandle, MessageQueuePermitDestroy))
-	{
-		if (heap_can_free(heapCapability, queueHandle) != 0)
-		{
-			return -EPERM;
-		}
-		queue_stop(queue);
-		return token_obj_destroy(heapCapability, handle_key(), queueHandle);
-	}
-	return -EINVAL;
+	// timeout unused?!
+	return unseal(queueHandle, MessageQueuePermitDestroy)
```

Comment:
Since every use of `unseal()` is then followed by `.and_then_int`, we might (skip the introduction of `and_then_int` and) build a more specialized combinator for local use?

```
int with_unsealed(CHERI_SEALED(MessageQueue *) handle, int permissions, auto &&then)
{
  return unseal(handle, permissions).either(then, [](int e) { return e; });
}
```
or perhaps
```
template <int permissions>
int with_unsealed(CHERI_SEALED(MessageQueue *) handle, auto &&then)
{
  return unseal(handle, permissions).either(then, [](int e) { return e; });
}
```